### DISCUSSION TITLE
How to `vips_text` with text color

### DISCUSSION THREAD

**Discussion by zhaohuxing** [libvips/libvips#2522] [Q&A] [answered]
Hi @jcupitt, I search issues known that `vips_text` and `vips_composite2` implement watermark text, but I need that implement watermark text with color. I don't find color args, how to write？😊

https://github.com/libvips/lua-vips/blob/master/example/watermark.lua
https://github.com/libvips/php-vips/issues/46

**Comment by jcupitt** [comment, marked as answer]
Hi @zhaohuxing,

`text` can make colour images. For example (using Python):

```python
#!/usr/bin/python3

import sys
import pyvips

image = pyvips.Image.new_from_file(sys.argv[1])
watermark = pyvips.Image.text("<span foreground='red'>hello</span>",
                              dpi=600, rgba=True)
image = image.composite(watermark, "over", x=300, y=300)
image.write_to_file(sys.argv[2])
```

**Reply by zhaohuxing** [reply]
Yes, I find the answer now! foreground support "#CCFFC" format, like!!! Thanks old sport @jcupitt
